packages/acceptance-tests/pkg-tests-specs/sources/commands/add.test.js

```diff
@@ -116,5 +117,56 @@ describe(`Commands`, () => {
         });
       }),
     );
+
+    test(`it should clean the cache when cache lives inside the project`, makeTemporaryEnv({
+      dependencies: {
+        [`no-deps`]: `1.0.0`,
+      },
+    }, async ({path, run, source}) => {
+      let code;
+      let stdout;
+      let stderr;
+
+      await run(`install`);
+
+      const preUpgradeCache = await readdir(`${path}/.yarn/cache`);
+
+      expect(preUpgradeCache.find(entry => entry.includes('no-deps-npm-1.0.0'))).toBeDefined();
+
+      ({ code, stdout, stderr } = await run(`add`, `no-deps@2.0.0`));
+
+      await expect({code, stdout, stderr}).toMatchSnapshot();
+
+      const postUpgradeCache = await readdir(`${path}/.yarn/cache`);
+
+      expect(postUpgradeCache.find(entry => entry.includes('no-deps-npm-1.0.0'))).toBeUndefined();
+      expect(postUpgradeCache.find(entry => entry.includes('no-deps-npm-2.0.0'))).toBeDefined();
+    }));
+
+    test(`it should not clean the cache when cache lives outside the project`, makeTemporaryEnv({
+      dependencies: {
+        [`no-deps`]: `1.0.0`,
+      },
+    }, async ({path, run, source}) => {
+      const sharedCachePath = await createTemporaryFolder();
+      const env = {
+        YARN_CACHE_FOLDER: sharedCachePath
```

Comment:
Fyi (not a blocker), it also can be done by writing a `.yarnrc.yml` file into `path`. We do this in a few other places (mainly for the plugins).